### DISCUSSION TITLE
4.0.0 was released 🚀🚀🚀

### DISCUSSION THREAD

**Discussion by pauldendulk** [Mapsui/Mapsui#2072] [General]
Thanks everybody who contributed! 🤩🤩🤩And special thanks to @inforithmics who did a ton of work on all areas .

We should hold back a bit with bigger changes for a while (not sure how long that is). There will be more bugs coming in the coming period. At some point we will need to work with separate branches, 1 for v4 maintenance and 1 for v5 development. If we create the branch we need to apply the fixes in two branches, so it is often easier to postpone that a bit. What is good timing depends on how many bugs come in, lets see.

It was one and a half year since the previous release. We have still a lot of changes planned. Let's hope the next major release will not take that long.

**Comment by pauldendulk** [comment]
The next week I try to spend time on going through the issues, update the docs, update the samples.

**Reply by vkotov** [reply]
You have done a great job!
Do you plan any service release for Mapsui 4.x before .NET 8.0 (November 2023) date?

Thanks

**Reply by janusw** [reply]
> Do you plan any service release for Mapsui 4.x before .NET 8.0 (November 2023) date?

I agree that this would be a good idea. @pauldendulk Maybe it is time to create a release branch and start thinking towards a version 4.0.1? One thing that I would like to see in there is #2099 (in order to fix package warnings when using Mapsui 4.0 with Svg.Skia 1.0).

@vkotov Are there any fixes that you would like to see in a service release?

**Reply by pauldendulk** [reply]
I have the holidays behind me and should have more time in the coming months. There are already quite a few fixes awaiting a new release.

**Reply by vkotov** [reply]
@janusw I'm waiting for #2159 (for now I successfully using @pauldendulk fixes of Avalonia touchandling)

**Comment by robloo** [comment]
Fantastic work!! It can't be overstated how useful this project is!

For those using WinUI3 or Avalonia it really is the only viable solution to allow mapping as well.

**Comment by pilotshamn** [comment]
This is an absolutely fantastic project that has facilitated multi platform development. Good work

**Comment by memsom** [comment]
When can we expect a new version? The map rotation bug that is now fixed in the master branch is affecting us and we could really use a timeline for when it will be available. I checked - we need #2128

**Reply by pauldendulk** [reply]
I can not promise anything because I don't have any time allotted for this project, it is just when I have some time available in my schedule, but I was hoping on 'in a few weeks', but my estimates have been wrong many times before.
